internal/services/compute/virtual_machine_scale_set.go

```diff
@@ -72,12 +72,7 @@ func ExpandVirtualMachineScaleSetHardwareProfile(input []interface{}) *compute.V
 
 func FlattenVirtualMachineScaleSetHardwareProfile(input *compute.VirtualMachineScaleSetHardwareProfile) []interface{} {
 	if input == nil || input.VMSizeProperties == nil {
-		return []interface{}{
```

Comment:
This causes diff when `hardware_profile` is not specified. saying updating these two values from 0 to nil, so returning an empty interface instead